### PR TITLE
Cleanup: PHP 5.3 support end

### PR DESCRIPTION
**Pull request by Ocramius**

### CODE REVIEW

**Comment by deeky666** [comment]
Finally! thx @Ocramius !

**Comment by driesvints** [comment]
Sweet!

**Comment by stof** [comment]
@deeky666 this will force you to make Doctrine ORM 2.4 an LTS for people being stuck with PHP 5.3 support

**Comment by githubjeka** [comment]
http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/getting-started.html#guide-assumptions

File: lib/Doctrine/ORM/Persisters/BasicEntityPersister.php [hunk 1]
```diff
@@ -558,19 +558,17 @@ protected function deleteJoinTableRecords($identifier)
...
     {
         $class      = $this->class;
-        $em         = $this->em;
-
         $identifier = $this->em->getUnitOfWork()->getEntityIdentifier($entity);
         $tableName  = $this->quoteStrategy->getTableName($class, $this->platform);
         $idColumns  = $this->quoteStrategy->getIdentifierColumnNames($class, $this->platform);
         $id         = array_combine($idColumns, $identifier);
-        $types      = array_map(function ($identifier) use ($class, $em) {
+        $types      = array_map(function ($identifier) use ($class) {
```
Comment (Taluu) [inline lib/Doctrine/ORM/Persisters/BasicEntityPersister.php]: how about `$class`, which is also an alias to `$this->class` ?

**Pull request merged**